### PR TITLE
[WIP] Remove knife bootstrap windows

### PR DESCRIPTION
**Pull request by vsingh-msys**
### Description

- In order to remove bootstrapping support from knife-windows related code has been refactored.
- Updated the README doc.

### Related PR

https://github.com/chef/knife-windows/pull/477

### Check List

- [ ] New functionality includes tests
- [x] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [x] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,10 +7,19 @@ Knife Windows Plugin
 This plugin adds additional functionality to the Chef Knife CLI tool for
 configuring / interacting with nodes running Microsoft Windows:
 
-* Bootstrap of nodes via the [Windows Remote Management (WinRM)](http://msdn.microsoft.com/en-us/library/aa384426\(v=VS.85\).aspx) or SSH protocols
 * Remote command execution using the WinRM protocol
 * Utilities to configure WinRM SSL endpoints on managed nodes
 
+## **Knife bootstrap windows ssh/winrm**
+
+Bootstrapping has been moved into chef-core and is no longer a part of knife-window
```
Comment (btm) [inline README.md]: > Core Chef now supports bootstrapping Windows systems without a knife plugin

File: README.md [hunk 2]
```diff
@@ -7,10 +7,19 @@ Knife Windows Plugin
...
+
+| Old knife-windows | New knife (chef-core) | Notes |
```
Comment (btm) [inline README.md]: > | knife-windows plugin | Chef Infra Client 15+ | Notes |

File: README.md [hunk 3]
```diff
@@ -7,10 +7,19 @@ Knife Windows Plugin
...
+|-----:|:-----------|:-----------|
+| knife windows bootstrap ssh |knife bootstrap -o ssh or knife bootstrap | Default is 'ssh'. | |
+| knife windows bootstrap winrm | knife bootstrap -o winrm| | |
```
Comment (btm) [inline README.md]: nitpicks:
* there is some whitespace inconsistency here to fix.
* I would drop the period after `Default is 'ssh'`
* It seems like there's an unused fourth column here

**Comment by btm** [comment]
we should probably keep `knife windows bootstrap winrm` and `knife windows bootstrap ssh` commands, but have them warn you to use `knife bootstrap ...` instead and then exit. that way it is immediately obvious the new command to type without searching the internet or looking at the docs.

File: README.md (hunk 3 above)
Comment (vsingh-msys) [reply README.md]: Thank you @btm for such pointers.

File: README.md (hunk 1 above)
Comment (vsingh-msys) [reply README.md]: Noted thanks

File: README.md (hunk 3 above)
Comment (marcparadise) [reply README.md]: s/(chef-core)/(train)

**Comment by vsingh-msys** [comment]
@btm restore the `knife windows bootstrap winrm` and `knife windows bootstrap ssh` commands. And added the following fatal warning
```
FATAL: *knife windows bootstrap winrm*
 Core Chef now supports bootstrapping Windows systems without a knife plugin
 
 Use 'knife bootstrap -o winrm' instead.
 
 For more detail https://github.com/chef/chef/blob/master/RELEASE_NOTES.md#knife-bootstrap

``

File: README.md (hunk 3 above)
Comment (vsingh-msys) [reply README.md]: Hey @marcparadise 
> s/(chef-core)/(train)

Could you please guide me on this? Thanks

**Comment by vsingh-msys** [comment]
@btm Should I created separate PR for README doc updates As I missed it?

File: README.md (hunk 3 above)
Comment (btm) [reply README.md]: @vsingh-msys he meant to replace "chef-core" with "train" but that was covered by a different change. `s/foo/bar/` is a regular expression substitution and is often used by developers to quickly say "replace foo with bar".

**Pull request merged**